src/test/java/ParserTest.java

```diff
@@ -0,0 +1,31 @@
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ParserTest {
+    @Test
```

Comment:
Very comprehensive JUnit testing overall! Test methods were descriptive, hence it was easy to understand what tests you were trying to run.